### PR TITLE
 Audio PR - Augmentation support [PreEmphasis Filter]

### PR DESCRIPTION
**Pull request by swetha097**
- Add Pre-Emphasis Filter support in rocAL

MIVisionX PR Link : [PR#1314](https://github.com/ROCm/MIVisionX/pull/1314)
NOTE: [PR#120](https://github.com/ROCm/rocAL/pull/120) to be merged before this PR - Then the actual file changes should reduce to 14

### CODE REVIEW

**Comment by kiritigowda** [comment]
@swetha097 -- please resolve the merge conflicts

**Comment by LakshmiKumar23** [comment]
@swetha097 can you resolve merge conflicts

File: rocAL/include/augmentations/audio_augmentations/node_preemphasis_filter.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    PreemphasisFilterNode() = delete;
+    void init(FloatParam *preemph_coeff, RocalAudioBorderType preemph_border);
+
+   protected:
+    void create_node() override;
+    void update_node() override;
+
+   private:
+    ParameterVX<float> _preemph_coeff;
+    constexpr static float PREEMPH_COEFF_RANGE[2] = {0.97, 0.97};
```
[the remark this answers is not in the document]
Comment (swetha097) [reply rocAL/include/augmentations/audio_augmentations/node_preemphasis_filter.h:42]: This is the default value given in DALI.

File: rocAL/include/augmentations/audio_augmentations/node_preemphasis_filter.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#pragma once
+#include "pipeline/graph.h"
+#include "pipeline/node.h"
+#include "parameters/parameter_factory.h"
+#include "parameters/parameter_vx.h"
+#include "rocal_api_types.h"
+
+class PreemphasisFilterNode : public Node {
+   public:
+    PreemphasisFilterNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs);
```
Comment (rrawther) [inline rocAL/include/augmentations/audio_augmentations/node_preemphasis_filter.h:32]: please use 4 spaces from the previous line here

File: rocAL/source/augmentations/audio_augmentations/node_preemphasis_filter.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    if ((status = vxGetStatus((vx_reference)_node)) != VX_SUCCESS)
+        THROW("Adding the vxExtRppPreemphasisFilter node failed: " + TOSTR(status))
+}
+
+void PreemphasisFilterNode::update_node() {
+    _preemph_coeff.update_array();
+}
+
+void PreemphasisFilterNode::init(FloatParam *preemph_coeff, RocalAudioBorderType preemph_border) {
+    _preemph_coeff.set_param(core(preemph_coeff));
```
Comment (rrawther) [inline rocAL/source/augmentations/audio_augmentations/node_preemphasis_filter.cpp:46]: I think preemph coef can be null pointer as per the definition. Please check the value before using it here

**Review by rrawther — CHANGES_REQUESTED** [review]
Please address the review comments

File: rocAL/source/augmentations/audio_augmentations/node_preemphasis_filter.cpp (hunk 3 above)
Comment (swetha097) [reply rocAL/source/augmentations/audio_augmentations/node_preemphasis_filter.cpp:48]: Added a CHECK

**Comment by swetha097** [comment]
> Please address the review comments

Addressed the comments @rrawther @LakshmiKumar23

**Review by rrawther — APPROVED** [review]

**Review by LakshmiKumar23 — APPROVED** [review]
Tests pass, can be merged after [MIVisionX PR1314](https://github.com/ROCm/MIVisionX/pull/1314) is merged

**Comment by kiritigowda** [comment]
@LakshmiKumar23 @rrawther -- we need to review the deps with mivisionx & rpp

**Comment by LakshmiKumar23** [comment]
> @LakshmiKumar23 @rrawther -- we need to review the deps with mivisionx & rpp

Yes, I have approved MIVisionX PR1314

**Comment by rrawther** [comment]
@swetha097 please add the MIVisionX version check and disable this feature until the next version of MIVisionX where audio feature will be enabled

**Comment by LakshmiKumar23** [comment]
> @swetha097 please add the MIVisionX version check and disable this feature until the next version of MIVisionX where audio feature will be enabled

@kiritigowda The CMake changes for Audio is done. Can we merge this?

**Comment by LakshmiKumar23** [comment]
@swetha097 please resolve merge conflicts

**Comment by swetha097** [comment]
> @swetha097 please resolve merge conflicts

Done

**Review by kiritigowda — APPROVED** [review]

**Pull request merged**